frontend/lib/framework/UniqueString.cpp

```diff
@@ -200,24 +201,49 @@ std::ostream& operator<<(std::ostream& os, const chpl::UniqueString& uStr) {
   return os;
 }
 
-//
-// Represent UniqueString occurrences as integer IDs in a table that the
-// Serializer builds up while serializing various entities. This reduces
-// the file size of generated serialized AST and helps with performance.
-//
 void UniqueString::serialize(Serializer& ser) const {
-  auto id = ser.cacheString(c_str(), length());
-  ser.write(id);
+  auto len = length();
+  if (len < ser.LONG_STRING_SIZE) {
+    // store the string inline here
+    uint8_t byte = len;
+    ser.writeByte(byte);
+    ser.writeData(c_str(), len);
+  } else {
+    // store the string through the long strings table
+    uint32_t uid = ser.cacheString(c_str(), len);
+    // won't work if the high bit is set
+    CHPL_ASSERT((uid >> 31) == 0);
+    // compute the uid in big-endian order with high bit set
+    uint32_t num = htonl(uid);
+    uint8_t bytes[4];
+    memcpy(&bytes[0], &num, sizeof(num));
+    bytes[0] |= 0x80; // set the high bit
```

Comment:
Would these kinds of operations and checks read more easily as a macro?

Reply:
What did you have in mind? IMO this is small enough and is contained to `UniqueString::serialize` and `UniqueString::deserialize`, so I'm not sure a macro would make it easier to read or maintain.

Reply:
I was just thinking something like ``SET_HIGH(bytes[0])``, to eliminate the need for the comment. Anyways, if we don't expect or intend to need this much elsewhere I think it's fine to leave as-is.